File: ios/Plugin/FilePickerPlugin.m
#import <Foundation/Foundation.h>
#import <Capacitor/Capacitor.h>

// Define the plugin using the CAP_PLUGIN Macro, and
// each method the plugin supports using the CAP_PLUGIN_METHOD macro.
CAP_PLUGIN(FilePickerPlugin, "FilePicker",
           CAP_PLUGIN_METHOD(chooseDirectory, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(accessDirectory, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(releaseDirectory, CAPPluginReturnPromise);
)

File: ios/Plugin/FilePickerPlugin.swift
import Foundation
import Capacitor

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(FilePickerPlugin)
public class FilePickerPlugin: CAPPlugin {
  @objc func chooseDirectory(_ call: CAPPluginCall) {
    let filePicker = FilePicker(call, self.bridge)
    filePicker.chooseDirectory()
  }
  
  @objc func accessDirectory(_ call: CAPPluginCall) {
    let filePicker = FilePicker(call, self.bridge)
    filePicker.accessDirectory()
  }
  
  @objc func releaseDirectory(_ call: CAPPluginCall) {
    let filePicker = FilePicker(call, self.bridge)
    filePicker.releaseDirectory()
  }
}
